### DISCUSSION TITLE
How to use beforeUpload in assetManager?

### DISCUSSION THREAD

**Discussion by andykramer** [GrapesJS/grapesjs#4510] [Q&A]
Hi,
for file size limitation reason, i would like to use the method `beforeUpload` of the `assteManager`.

How is the correct way to use this methode?

I try to use it like this:
```
var editor = grapesjs.init({
  assetManager: {
    multiUpload: true,
    beforeUpload: (files) => {
       console.log(files);
    },
  },
});
```
Thanks

**Comment by artf** [comment]
You can return `false` in `beforeUpload` in order to cancel the upload.

**Reply by andykramer** [reply]
Hello artf
I know, but `beforeUpload ` is never called.
I try something like this: 
```
assetManager: {
  multiUpload: true,
  beforeUpload: (files) => {
   console.log(files);
   var maxFilesize = 3300000;
   var errorMessage = "";
   var error = false;

   for (let i = 0; i < files.length; i++) {
     if (files[i].size > maxFilesize) {
       errorMessage += "Image: " + files[i].name + " is too large. (" + files[i].size + ")\n";
       error = true;
     }
   }
   if (!error) {
     return true;
   }
   alert(errorMessage);
   return false;
 },
},
```
But this does not work! not even "console.log(files)"

**Reply by artf** [reply]
Did you configure properly the uploader or you're using the default `embedAsBase64` option? In that case `beforeUpload` triggers only before the real upload of files.
